pkg/gatewayserver/gatewayserver.go

```diff
@@ -638,6 +644,74 @@ func (gs *GatewayServer) handleUpstream(conn connectionEntry) {
 	}
 }
 
+// UpdateConnectionStats updates the stats for a single gateway connection.
+func (gs *GatewayServer) UpdateConnectionStats(ctx context.Context, conn *io.Connection) error {
+	stats := &ttnpb.GatewayConnectionStats{}
+	ct := conn.ConnectTime()
+	stats.ConnectedAt = &ct
+	stats.Protocol = conn.Frontend().Protocol()
+	if s, t, ok := conn.StatusStats(); ok {
+		stats.LastStatusReceivedAt = &t
+		stats.LastStatus = s
+	}
+	if c, t, ok := conn.UpStats(); ok {
+		stats.LastUplinkReceivedAt = &t
+		stats.UplinkCount = c
+	}
+	if c, t, ok := conn.DownStats(); ok {
+		stats.LastDownlinkReceivedAt = &t
+		stats.DownlinkCount = c
+	}
+	if min, max, median, count := conn.RTTStats(); count > 0 {
+		stats.RoundTripTimes = &ttnpb.GatewayConnectionStats_RoundTripTimes{
+			Min:    min,
+			Max:    max,
+			Median: median,
+			Count:  uint32(count),
+		}
+	}
```

Comment:
This is exactly the same as in the gRPC getter. Please make this a func in `*io.Connection`